support/ppPrint.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#ifdef SPDLOG_ENABLED
+  #include "spdlog/spdlog.h"
+  #include <spdlog/fmt/bundled/printf.h>
+#endif
+
+#include <Kokkos_Core.hpp>
+#include "ppMacros.h"
+
+namespace pumipic {
+
+  #if defined(__CUDA_ARCH__) || defined(__HIP_DEVICE_COMPILE__) || defined(__SYCL_DEVICE_ONLY__)
+    #define ACTIVE_GPU_EXECUTION
+  #endif
+
+  inline FILE* pp_stdout = stdout;
+  inline FILE* pp_stderr = stderr;
```

Comment:
I think this could potentially cause issues since this is a header file these will potentially be set in each translation unit. If we actually want global state here, it probably needs to be extern with the storage in a cpp file. Could this be stored in the pumipic library object instead?

Global state has bitten me badly in the past see: https://github.com/SCOREC/core/pull/433 ... getting rid of the singleton required touching almost every file in the project.

So, I'd like to avoid if we can.